src/app/components/Blocks/index.tsx

```diff
@@ -0,0 +1,129 @@
+import { Link as RouterLink } from 'react-router-dom'
+import { useTranslation } from 'react-i18next'
+import Link from '@mui/material/Link'
+import formatDistanceStrict from 'date-fns/formatDistanceStrict'
+
+import { RuntimeBlockList } from '../../../oasis-indexer/generated/api'
+import { VerticalProgressBar } from '../../components/ProgressBar'
+import { Table, TableCellAlign, TableColProps } from '../../components/Table'
+import { trimLongString } from '../../utils/trimLongString'
+import { intlDateFormat } from '../../utils/dateFormatter'
+
+type BlocksProps = RuntimeBlockList & {
+  isLoading: boolean
+  limit: number
+  verbose?: boolean
+  pagination?: boolean
+}
+
+const gasLimit = 1000000 // TODO temporary value
+
+export const Blocks = (props: BlocksProps) => {
+  const { isLoading, blocks, verbose, pagination, limit } = props
+  const { t } = useTranslation()
+
+  const tableColumns: TableColProps[] = [
+    { content: t('common.table.fill') },
+    { content: t('common.table.block'), align: TableCellAlign.Right },
+    verbose ? { content: t('common.table.timestamp') } : undefined,
```

Comment:
```suggestion
    ...(verbose ? [{ content: t('common.table.timestamp') }] : []),
```

This would be better, and getting rid of `filter` on array, which would reduce complexity.

Reply:
Actually I started out that way, but then I decided that enumerating all the columns that way is too much of PITA, so I want for the filter. But it doesn't matter either way, I can revert to the previous method.

Reply:
I agree it's not readable, but it is be more efficient. If you want you can add an utility for it.

Reply:
OK I changed it as suggested